vault/logical_system_activity.go

```diff
@@ -216,5 +216,13 @@ func (b *SystemBackend) handleActivityConfigUpdate(ctx context.Context, req *log
 	// Set the new config on the activity log
 	a.SetConfig(ctx, config)
 
+	if config.Enabled == "disable" {
+		return &logical.Response{
```

Comment:
Could we be more specific about doing this only one enable->disable changes?

I think this would appear even if we changed the retention period while activity log was disabled.

Another thing to think about is reset back to "default" when running on OSS, which is also disabled.

Reply:
> Could we be more specific about doing this only on enable->disable changes?

Good catch. Updated.

> Another thing to think about is reset back to "default" when running on OSS, which is also disabled.

Can you elaborate a bit more on this?

Reply:
> > Another thing to think about is reset back to "default" when running on OSS, which is also disabled.

> Can you elaborate a bit more on this?

I imagine he just means the following. Activity log is disabled by default on OSS (enabled by default on ENT). If a user is running OSS with activity log set to enable and then wants to restore to default settings, it will disable activity log (thus deleting data which should trigger a notification)

Reply:
I've pushed an update to handle when enabled -> default. Since this is in the OSS repo, I imagine when we port this to ENT, we'll have this removed. Is that the correct line of thinking? Or is there some other way to check?

Reply:
Your line of thinking is correct, but I think the implementation might be different. The pattern I've seen so far when the same function signature needs different behavior is to use build tags (so when we compile for OSS we get one version of the function and when we compile for ENT we get a different version of the function). Since this is an OSS PR I'm not sure that I should comment a specific ent example here, so I'll DM you.

@ncabatoff seems to have quite a bit of experience with this, so it might be good to get his thoughts

Reply:
core_util.go has activityLogEnabledDefault, which is false on OSS builds and true on Enterprise builds; it's probably better to use that value than to have other code that differs between OSS and ENT.